docs/onnx-conversion.md

```diff
@@ -1,75 +1,251 @@
+# Load the ONNX model
+onnx_model = onnx.load(model_path)
+model_metadata = onnx_model.metadata_props
+print(model_metadata)
+
+model_type = next((prop.value for prop in model_metadata if prop.key == 'model_type'), None)
+num_heads = next((int(prop.value) for prop in model_metadata if prop.key == 'num_heads'), None)
+hidden_size = next((int(prop.value) for prop in model_metadata if prop.key == 'hidden_size'), None)
+```
+
+The above captures the metadata that we stored in the previous section for [converting the model to ONNX](#adding-metadata). 
+
+### Run Optimizer on ONNX Model
+
+As the final step for completing the optimization, we will be providing all the data to the ONNX optimizer:
+
+```python
+# Optimize the model
 optimized_model = optimizer.optimize_model(
-    'splade-pp-ed-optimized.onnx',
-    model_type='bert',
-    num_heads=12,
-    hidden_size=768
+    model_path,
+    model_type=model_type,
+    num_heads=num_heads,
+    hidden_size=hidden_size
 )
 
-# Optional: convert model to float16 (if not in fp16)
-# optimized_model.convert_float_to_float16()
-
-optimized_model.save_model_to_file('splade-pp-ed-optimized-fp16.onnx')
+# Optionally, we can convert to FP16 as well (not done in the script):
+optimized_model.convert_float_to_float16()
 ```
 
-## Inference with ONNX Runtime
-The following sections will describe how to run inference with ONNX Runtime. The steps are as follows:
+### Save Optimized Model
+
+Lastly, after optimizing the model, we just need to save it as an optimized ONNX model:
+
 ```python
-import onnxruntime
+model_name = model_path.split(".")[0]
+optimized_model_path = f'{model_name}-optimized.onnx'
+optimized_model.save_model_to_file(optimized_model_path)
+print(f"ONNX model optimization successful. Saved to {optimized_model_path}")
+```
+
+## Running Inference
+
+As the final step of this tutorial, we can use the ```onnxruntime.InferenceSession``` module for a simple inference session. 
+
+### Run the End to End Inference
+
+Loading and running is done easily with argparse in the following script:
+```
+src/main/python/onnx/run_onnx_model_inference.py
+```
 
-splade = onnxruntime.InferenceSession('path-to-splade-pp-ed-optimized-fp16.onnx')
-tokenizer = AutoTokenizer.from_pretrained(model_type_or_dir)
+For this example, we will continue with the SPLADE++ Ensemble Distil model.
 
-inputs = tokenizer("This is a sample input", return_tensors="np")
-splade.run(None , {"input_ids": inputs['input_ids'], 
-                   "token_type_ids": inputs['token_type_ids'], 
-                   "attention_mask": inputs['attention_mask']})
+To run the script for running inference, run the following sequence of commands:
+```bash
+# Begin by going to the appropriate directory
+cd src/main/python/onnx
+# Now run the script
+python3 run_onnx_model_inference.py --model_path models/splade-cocondenser-ensembledistil-optimized.onnx \
+                                    --model_name naver/splade-cocondenser-ensembledistil
```

Comment:
maybe an explanation here that we get the model nome to use the tokenizer? 
Another possibility is using the model name to load the hf version and then do a comparison between both to check that things are ok (always reassuring to someone using this for the first time)

Reply:
Sounds good. I can add more details for this